torch/utils/_sympy/reference.py

```diff
@@ -76,8 +76,25 @@ def truediv(a, b):
 
     @staticmethod
     def div(a, b):
+        if a.is_integer:
+            return ReferenceAnalysis.truncdiv(a, b)
+        else:
+            return ReferenceAnalysis.truediv(a, b)
```

Comment:
nyeh, I don't understand. If b isn't an integer, you'll still end up with truediv no?

Reply:
Actually, now that I think about it, it's just when `a` is not an integral that dispatches to `div`, so `div` should be just `truediv` when on Inductor IR. cc @peterbell10